monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala

```diff
@@ -369,6 +394,74 @@ object Consumer {
     }
   }
 
+  /** Implementation for [[Consumer.partitionParallel]] */
+  private final class PartitionParallelConsumer[A](nrPartitions: Int, overflowStrategy: OverflowStrategy.Synchronous[A], partition: A => Int, consumer: PartitionedConsumer[A]) extends Consumer[A, Unit] {
+    override def createSubscriber(cb: Callback[Unit], s: Scheduler): (Subscriber[A], AssignableCancelable) = {
+
+      val errors = scala.collection.mutable.ArrayBuffer.empty[Throwable]
+
+      val obs = PublishSubject[A]()
+      val pub = obs.whileBusyBuffer(overflowStrategy)
+      val isUpstreamDone = AtomicBoolean(false)
+
+      val refCount = RefCountCancelable { () =>
+        errors.synchronized {
```

Comment:
When this happens, it means that all subscribers have already received their `onComplete` or `onError`, hence you don't need to synchronize on `errors`, because no other `onError` events can happen after this is triggered.